pkg/apis/core/validation/validation.go

```diff
@@ -3019,19 +3033,27 @@ func validateHandler(handler commonHandler, fldPath *field.Path) field.ErrorList
 			allErrors = append(allErrors, validateGRPCAction(handler.GRPC, fldPath.Child("grpc"))...)
 		}
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodLifecycleSleepAction) && handler.Sleep != nil {
```

Comment:
Don't check the gate here.  We checked it at API admit time.  If the field is set, it's because it was allowed to be set, and we must validate it.